docs/en/02_Developer_Guides/15_Customising_the_Admin_Interface/How_Tos/Extend_CMS_Interface.md

```diff
@@ -642,6 +642,146 @@ class MyGallery extends React.Component {
+```
+
+Alternatively, you can also register your own reducer to injector, particularly if you're building your own module.
+
+```js
+Injector.reducer.register('myCustom', MyCustomerReducer);
+```
+
+#### Structuring a transformer
+
+We utilise currying to supply utilities which your transformer may require to handle the transformation.
+- `originalReducer` - reducer callback which the transformer is customising, this will need to be called in most cases. This will also callback other transformations down the chain of execution, not calling this will break the chain.
+- `globalState` - state of the global redux store, there may be data outside the current scope in the reducer which you may need to help determine the transformation.
+- `state` - current state of the current scope, this is what should be used to form the new State.
+- `type` - the action to fire, like in any reducer in redux, this helps determine if your transformer should do anything
+- `payload` - the new data sent with the action to mutate the redux store.
+
+```js
+const MyReducerTransformer = (originalReducer) => (globalState) => (state, { type, payload }) => {
+  switch (type) {
+    case 'MY_ACTION': {
+      // recommended to call and return the originalReducer with the payload changed by the transformer
+      /* return action to call here; */
+    }
+    
+    case 'ANOTHER_ACTION_TRANSFORM': {
+      // could omit the originalReducer to enforce your change or cancel the originalREducer's change
+    }
+
+    default: {
+      // it is important to return the originalReducer with original redux parameters.
+      return originalReducer(state, { type, payload });
+    }
+  }
+}
+```
+
+#### A basic transformation
+
+We manipulate the payload data to show something else instead of "Files" before calling the originalReducer
+
+```js
+const MyReducerTransformer = (originalReducer) => (globalState) => (state, { type, payload }) => {
+  switch (type) {
+    case 'MY_ACTION': {
+      return originalReducer(state, {
+        type,
+        payload: {
+          breadcrumbs: payload.breadcrumbs.map((crumb) => (
+            (crumb.text === 'Files')
+              ? { ...crumb, text: 'Custom Files' }
+              : crumb
+          )),
+        },
+      });
+    }
+  }
+};
+```
+
+#### Using the globalState
+
+Accessing the globalState is easy, as it is passed in as part of the curried functions definition.
+
+```js
+export default (originalReducer) => (globalState) => (state, { type, payload }) => {
+  const baseUrl = globalState.config.baseUrl;
+  
+  switch (type) {
+    /* ... cases here ... */
+  }
+}
+```
+
+#### Setting a different initial state
+
+We can easily define a new initial state by providing the `state` param with a default value,
+ this does not work if an earlier transformer in the chain has defined an initial state already.
+
+```js
+const MyReducerTransformer = (originalReducer) => () => (state = { myCustom: 'initial state here'}, { type, payload }) => {
```

Comment:
That's a bit limiting, isn't it? It'll only set the `myCustom` key if the *entire* state tree has been undefined at this point. What happens when two transformers want to add initial state? Can't we just instruct the reader to set the initial state in the body?